src/containers/App.jsx

```diff
@@ -37,10 +46,12 @@ class App extends Component {
   componentWillMount() {
     // retrieve token from cookie
     const token = getToken();
-    if (token === null || tokenIsValid(token) === false || isFellow(token) === false) {
+    const tokenInfo = decodeToken(token);
+    if (token === null || tokenIsValid(tokenInfo) === false || isFellow(tokenInfo) === false) {
```

Comment:
I will like to suggest a more concise way for this
`if (!token || !tokenIsValid(tokenInfo) || !isFellow(tokenInfo)) {`